orchestrators/ProcurementOperations/src/Services/DeterministicVendorScorer.php

```
@@ -0,0 +1,198 @@
+
+        return new VendorRecommendationScoredCandidate(
+            vendorId: $candidate->vendorId,
+            vendorName: $candidate->vendorName,
+            fitScore: $score,
+            confidenceBand: VendorRecommendationScoredCandidate::confidenceBandFor($score),
+            recommendedReasonSummary: $reasons[0],
+            deterministicReasons: $reasons,
+            warningFlags: array_values(array_unique($warningFlags)),
+            warnings: array_values(array_unique($warnings)),
+        );
+    }
```

PHPMD (2.15.0)

[warning] The method scoreCandidate() has a Cyclomatic Complexity of 19. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)